app/labor/rate_limit_checker.rb

```diff
@@ -48,13 +39,26 @@ def ping_admins
     RateLimitCheckerWorker.perform_async(user.id, action)
   end
 
+  def check_comment_creation_limit
+    user.comments.where("created_at > ?", 30.seconds.ago).size > 9
+  end
+
+  def check_published_article_creation_limit
+    user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
+  end
+
+  def check_image_upload_limit
+    Rails.cache.read("#{user.id}_image_upload").to_i > 9
+  end
+
+  def check_follow_account_limit
+    user_today_follow_count > self.class.daily_account_follow_limit
+  end
```

Comment:
why keep them all public? I would even move `ping_admins` in the private section